packages/meditrak-app-server/src/routes/sync/PushChangesRoute/PushChangesRoute.ts

```diff
@@ -35,27 +43,38 @@ export class PushChangesRoute extends Route<PushChangesRequest> {
 
     const surveyResponses = [];
     const surveyImages: { id: string; data: string }[] = [];
+    let waitForAnalyticsRebuild = false;
 
-    for (const { action, payload } of changes) {
+    for (const { action, payload, ...rest } of changes) {
       if (!VALID_ACTIONS.includes(action)) {
         throw new ValidationError(`${action} is not a supported change action`);
       }
 
-      if (action === 'SubmitSurveyResponse') {
+      if (action === SUBMIT_SURVEY_RESPONSE) {
         const translatedPayload = await translateSurveyResponseObject(this.req.models, payload);
         const validatedSurveyResponse = await validateSurveyResponseObject(
           this.req.models,
-          translatedPayload,
+          translatedPayload.survey_response || translatedPayload, // LEGACY: v1 and v2 allow survey_response object, v3 should deprecate this
+        );
+        await upsertCreatedData(this.req.models, validatedSurveyResponse);
+        const surveyResponseWithPopulatedData = await populateData(
+          this.req.models,
+          validatedSurveyResponse,
         );
-        surveyResponses.push(validatedSurveyResponse);
+        surveyResponses.push(surveyResponseWithPopulatedData);
+        waitForAnalyticsRebuild = rest.waitForAnalyticsRebuild || waitForAnalyticsRebuild;
       } else {
         const validatedPayload = addSurveyImageValidator.validateSync(payload);
         surveyImages.push(validatedPayload);
       }
     }
 
     // Submit survey responses
-    await this.req.ctx.services.central.createSurveyResponses(surveyResponses);
+    await this.req.ctx.services.central.upsertSurveyResponses(surveyResponses);
+    if (waitForAnalyticsRebuild) {
+      const { database } = this.req.models;
+      await AnalyticsRefresher.refreshAnalytics(database);
```

Comment:
Is it fine to run refreshAnalytics() after all survey responses have been upserted? Before we do it for each survey response.

Reply:
Hmmm can we actually remove this functionality? It's only used by PSSS, which won't be using this route.

Later on when we clean up the original route in central-server, we will need to decide what to do with PSSS...